File: Grunt/Grunt/Models/HaloInfinite/Appearance.cs
// <copyright file="Appearance.cs" company="Den Delimarsky">
// Developed by Den Delimarsky.
// Den Delimarsky licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
// The underlying API powering Grunt is managed by 343 Industries and Microsoft. This wrapper is not endorsed by 343 Industries or Microsoft.
// </copyright>

namespace OpenSpartan.Grunt.Models.HaloInfinite
{
    /// <summary>
    /// Player appearance configuration.
    /// </summary>
    [IsAutomaticallySerializable]
    public class Appearance
    {
        /// <summary>
        /// Gets or sets the last modified date.
        /// </summary>
        public APIFormattedDate? LastModifiedDateUtc { get; set; }

        /// <summary>
        /// Gets or sets the service tag.
        /// </summary>
        public string? ServiceTag { get; set; }

        /// <summary>
        /// Gets or sets the path for the action pose.
        /// </summary>
        public string? ActionPosePath { get; set; }

        /// <summary>
        /// Gets or sets the path for the stance.
        /// </summary>
        public string? StancePath { get; set; }

        /// <summary>
        /// Gets or sets the path to the backdrop image for the Spartan ID.
        /// </summary>
        public string? BackdropImagePath { get; set; }

        /// <summary>
        /// Gets or sets the emblem.
        /// </summary>
        public Emblem? Emblem { get; set; }

        /// <summary>
        /// Gets or sets the intro emote path.
        /// </summary>
        public string? IntroEmotePath { get; set; }
    }
}

File: src/node/src/models/halo-infinite/customization/appearance.ts
/**
 * Spartan body configuration.
 */
export interface SpartanBody {
  /** When last modified (ISO 8601) */
  LastModifiedDateUtc?: string;
  /** Left arm prosthetic path */
  LeftArm?: string;
  /** Right arm prosthetic path */
  RightArm?: string;
  /** Left leg prosthetic path */
  LeftLeg?: string;
  /** Right leg prosthetic path */
  RightLeg?: string;
  /** Body type identifier */
  BodyType?: string;
  /** Voice number */
  Voice?: number;
  /** Voice path */
  VoicePath?: string;
}

/**
 * Player appearance configuration.
 */
export interface Appearance {
  /** When last modified (ISO 8601) */
  LastModifiedDateUtc?: string;
  /** Service tag (4 characters) */
  ServiceTag?: string;
  /** Intro emote path */
  IntroEmotePath?: string;
  /** Stance path */
  StancePath?: string;
  /** Emblem configuration */
  Emblem?: Emblem;
  /** Backdrop image path */
  BackdropImagePath?: string;
  /** Action pose path */
  ActionPosePath?: string;
  /** Player title path */
  PlayerTitlePath?: string;
}

/**
 * Emblem configuration.
 */
export interface Emblem {
  /** Emblem path */
  EmblemPath?: string;
  /** Emblem configuration ID */
  ConfigurationId?: number;
  /** Alternative path reference */
  Path?: string;
  /** Location identifier */
  LocationId?: number;
}

/**
 * Complete player customization data.
 */
export interface CustomizationData {
  /** Spartan body configuration */
  SpartanBody?: SpartanBody;
  /** Appearance settings */
  Appearance?: Appearance;
  /** Armor cores */
  ArmorCores?: import('./cores').ArmorCoreCollection;
  /** Weapon cores */
  WeaponCores?: import('./cores').WeaponCoreCollection;
  /** Vehicle cores */
  VehicleCores?: import('./cores').VehicleCoreCollection;
  /** AI cores */
  AiCores?: import('./cores').AiCoreContainer;
}

/**
 * Appearance customization container.
 */
export interface AppearanceCustomization {
  /** Service tag */
  ServiceTag?: string;
  /** Appearance configuration */
  Appearance?: Appearance;
}
